packages/@react-spectrum/label/src/ReadOnlyField.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames} from '@react-spectrum/utils';
+import {FocusRing} from '@react-aria/focus'; 
+import {mergeProps} from '@react-aria/utils';
+import React, {RefObject, useCallback} from 'react';
+import {SpectrumFieldProps} from '@react-types/label';
+import styles from '@adobe/spectrum-css-temp/components/textfield/vars.css';
+import {useFormProps} from '@react-spectrum/form';
+import {useHover} from '@react-aria/interactions';
+import {useLayoutEffect} from '@react-aria/utils';
+
+
+function ReadOnlyField(props: SpectrumFieldProps, ref: RefObject<HTMLInputElement | HTMLTextAreaElement>) {
+  props = useFormProps(props);
+  let {
+    isDisabled,
+    readOnlyText,
+    inputProps,
+    autoFocus
+  } = props;
+
+  let {hoverProps, isHovered} = useHover({isDisabled});
+
+  let onHeightChange = useCallback(() => {
+    let input = ref.current;
+    let prevAlignment = input.style.alignSelf;
+    input.style.alignSelf = 'start';
+    input.style.height = `${input.scrollHeight}px`;
+    input.style.alignSelf = prevAlignment;
+  }, [ref]);
+
+  useLayoutEffect(() => {
+    if (ref.current) {
+      onHeightChange();
+    }
+  }, [onHeightChange, ref]);
```

Comment:
This should have `readOnlyText` in the dep list. Make a controlled readOnly story that passes isReadOnly to this component: https://github.com/adobe/react-spectrum/blob/373244834584954dd407853c12e2bc4d68d96162/packages/%40react-spectrum/textfield/stories/TextArea.stories.js#L164-L172